### ISSUE TITLE
llvm-strip utility strips debuglink sections by default whereas GNU strip does not

### ISSUE THREAD

**Issue by kraj** [llvm/llvm-project#57407]
I ran into a problem where debug info was not being loaded by gdb. In yocto we split the debug info and keep them separate.
so it turns out to be that when we use `llvm-strip` for providing `STRIP` it ends up eating `.gnu_debuglink` section as well which contains the link to separated out debug info file and hence the debugger can not find the link any more. This is seen with clang-14/clang-15 and trunk too.

Here is a testcase

```shell
set -x
#CROSS=x86_64-yoe-linux-
#TC=llvm-
rm -rf a.out*
clang -g -xc /dev/null -ffreestanding -shared -o a.out
${CROSS}${TC}objcopy --version
${CROSS}${TC}objcopy --only-keep-debug a.out a.out.dbg
${CROSS}${TC}objcopy --add-gnu-debuglink=a.out.dbg a.out
#${CROSS}${TC}strip --remove-section=.note -o a.out.stripped a.out
${CROSS}${TC}strip -o a.out.stripped a.out
#${CROSS}${TC}objcopy --strip-debug a.out a.out.stripped
${CROSS}readelf -wk a.out.stripped
```

bad 

```shell
TC=llvm- sh build.sh
```
good
```shell
sh build.sh
```

**Comment by kraj** [comment]
@MaskRay FYI

**Comment by kraj** [comment]
workaround for now is to add `--keep-section=.gnu_debuglink` explicitly to `llvm-strip` cmdline. But I think it would be good if it behaved like GNU strip here.

**Comment by felixkellenbenz** [comment]
Hey is anyone working on this issue? if not could you please assign it to me ?

**Comment by felixkellenbenz** [comment]
Hey, my approach would be to add the `.gnu_debuglink` file to  `KeepSection` of the `CommonConfig` by default. Maybe someone could tell me if that's viable or if I should think of another approach. Thanks!

**Comment by jh7370** [comment]
> Hey, my approach would be to add the `.gnu_debuglink` file to `KeepSection` of the `CommonConfig` by default. Maybe someone could tell me if that's viable or if I should think of another approach. Thanks!

I suspect this functionality wants to be specific to the ELF file format, so adding it to `CommonConfig` wouldn't be the right approach. Some questions to answer:
1) Should this impact llvm-objcopy as well as llvm-strip (I suspect it should)?
2) Is the section considered a "debug" section? In other words, if you used the --strip-debug option, would it be removed by GNU strip or objcopy?

If the answer to both questions is yes, then it probably makes sense to add the section name to the list of section names specified in `isDebugSection` in `ELFObjcopy.cpp` (see https://github.com/llvm/llvm-project/blob/295edaab13326d842f8298ad07e9a5f9e5d3b131/llvm/lib/ObjCopy/ELF/ELFObjcopy.cpp#L55).

**Comment by felixkellenbenz** [comment]
Thanks for your answer, I will look into this as soon as possible.

**Comment by felixkellenbenz** [comment]
Hey,

the issue was fixed by #78919, could someone close it ?
